File: example/request.py
import requests
import json

url = 'http://URI/api/'
token = "123"

def insert(name, email, username, password):
    payload = {
        "name": name,
        "email": email,
        "username": username,
        "password": password
    }

    header = {"content-type": "application/json",
              "Authorization": token
              }

    response = requests.post(url+"new/", data=json.dumps(
        payload), headers=header, verify=True)
    r_json = response.text

    print(response)
    print(r_json)


def search(username, password):
    payload = {
        "username": username,
        "password": password
    }

    header = {"content-type": "application/json",
              "Authorization": token
              }

    response = requests.post(url+"/search/", data=json.dumps(
        payload), headers=header, verify=True)
    r_json = response.text

    print(response)
    print(r_json)


def update(name, email, username, password):
    payload = {
        "name": name,
        "email": email,
        "username": username,
        "password": password
    }

    header = {"content-type": "application/json",
              "Authorization": token
              }

    response = requests.put(url+"/update/", data=json.dumps(
        payload), headers=header, verify=True)
    r_json = response.text

    print(response)
    print(r_json)


# Examples - Uncomment to try

#insert("Full Name", "email@domain.com", "john", "doe")
#search("john", "doe")
#update("Full Name", "email@domain.com", "john", "doe")
#search("john", "doe")

File: example/request.php
<?php

# Search example, try with "new" and "update" (see documentation)
$url = 'http://URI/api/search/';
$token = '123';

$array = array(
    'username'=> 'john',
    'password'=> 'doe'
);

$json = json_encode( $array );

$ch = curl_init( $url );
curl_setopt( $ch, CURLOPT_CUSTOMREQUEST, 'POST' ); # PUT to update
curl_setopt( $ch, CURLOPT_POSTFIELDS, $json );
curl_setopt( $ch, CURLOPT_RETURNTRANSFER, true );
curl_setopt( $ch, CURLOPT_HTTPHEADER, array(
    'Content-Type: application/json',
    'Authorization: ' . $token,
    'Content-Length: ' . strlen( $json ) )
);

$obj = json_decode( curl_exec( $ch ) );

print_r( $obj );